src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -388,7 +388,9 @@ public static bool DoesFragmentConditionMatch(ExecutionContext context, string f
 
         public static IDictionary<string, Field> SubFieldsFor(ExecutionContext context, IGraphType fieldType, Field field)
         {
-            if (!(fieldType is IObjectGraphType) || !field.SelectionSet.Selections.Any())
```

Comment:
I'm not sure why there was a restriction to only IObjectGraphType. I changed this to simply verify that there are selections available.